File: Spa/src/domain/blog/Sort.ts
export declare type SortType = {
  value: number
  title: string
}

export enum SortEnum {
  DATE_ASC = 0,
  DATE_DESC = 1,
  TITLE_ASC = 2,
  TITLE_DESC = 3,
  CLAP_ASC= 4,
  CLAP_DESC= 5,
}

export const sortList: SortType[] = [
  {
    title: 'Date Asc',
    value: SortEnum.DATE_ASC 
  },
  {
    title: 'Title Asc',
    value: SortEnum.TITLE_ASC 
  },
  {
    title: 'Clap Asc',
    value: SortEnum.CLAP_ASC 
  },
  {
    title: 'Date Desc',
    value: SortEnum.DATE_DESC 
  },
  {
    title: 'Title Desc',
    value: SortEnum.TITLE_DESC 
  },
  {
    title: 'Clap Desc',
    value: SortEnum.CLAP_DESC 
  },
]

File: Api1/domain/blog/sort.py
from typing import Dict
from sqlalchemy import asc, desc


SortMap: Dict = {
        '0': {  # default
            "attr": "createdDate",
            "order": asc
            },
        '1': {
            "attr": "createdDate",
            "order": desc
            },
        '2': {
            "attr": "title",
            "order": asc
            },
        '3': {
            "attr": "title",
            "order": desc
            },
        '4': {
            "attr": "clap",
            "order": asc
            },
        '5': {
            "attr": "clap",
            "order": desc
            },
        }



